packages/protocol/test/common/gascurrencywhitelist.ts

```diff
@@ -1,40 +1,88 @@
+const MockSortedOracles: MockSortedOraclesContract = artifacts.require('MockSortedOracles')
+const StableToken: StableTokenContract = artifacts.require('StableToken')
+
+// @ts-ignore
+StableToken.numberFormat = 'BigNumber'
 
 contract('FeeCurrencyWhitelist', (accounts: string[]) => {
-  let feeCurrencyWhitelist: FeeCurrencyWhitelistInstance
+    let feeCurrencyWhitelist: FeeCurrencyWhitelistInstance
+    let registry: RegistryInstance
+    let stableToken: StableTokenInstance
+    let mockSortedOracles: MockSortedOraclesInstance
 
-  const aTokenAddress = '0x000000000000000000000000000000000000ce10'
+    const nonOwner = accounts[1]
 
-  const nonOwner = accounts[1]
+    const decimals = 18
+    const SECONDS_IN_A_WEEK = 604800
+    const goldAmountForRate = new BigNumber('0x10000000000000000')
+    const stableAmountForRate = new BigNumber(2).times(goldAmountForRate)
+    beforeEach(async () => {
+        stableToken = await StableToken.new()
+        registry = await Registry.new()
 
-  beforeEach(async () => {
-    feeCurrencyWhitelist = await FeeCurrencyWhitelist.new()
-    await feeCurrencyWhitelist.initialize()
-  })
+        await stableToken.initialize(
+            'Celo Dollar',
+            'cUSD',
+            decimals,
+            registry.address,
+            fixed1,
+            SECONDS_IN_A_WEEK,
+            [],
+            [],
+            'Exchange'
+        )
 
-  describe('#initialize()', () => {
-    it('should have set the owner', async () => {
-      const owner: string = await feeCurrencyWhitelist.owner()
-      assert.equal(owner, accounts[0])
+        mockSortedOracles = await MockSortedOracles.new()
+        await registry.setAddressFor(CeloContractName.SortedOracles, mockSortedOracles.address)
+        feeCurrencyWhitelist = await FeeCurrencyWhitelist.new()
+        await feeCurrencyWhitelist.initialize(registry.address)
     })
 
-    it('should not be callable again', async () => {
-      await assertRevert(feeCurrencyWhitelist.initialize())
-    })
-  })
+    describe('#initialize()', () => {
+        it('should have set the owner', async () => {
+            const owner: string = await feeCurrencyWhitelist.owner()
+            assert.equal(owner, accounts[0])
+        })
 
-  describe('#addToken()', () => {
-    it('should allow the owner to add a token', async () => {
-      await feeCurrencyWhitelist.addToken(aTokenAddress)
-      const tokens = await feeCurrencyWhitelist.getWhitelist()
-      assert.sameMembers(tokens, [aTokenAddress])
+        it('should not be callable again', async () => {
+            await assertRevert(feeCurrencyWhitelist.initialize(registry.address))
+        })
     })
 
-    it('should not allow a non-owner to add a token', async () => {
-      await assertRevert(feeCurrencyWhitelist.addToken(aTokenAddress, { from: nonOwner }))
+    describe('#addToken()', () => {
+        it('should not allow to add a token with an invalid oracle price', async () => {
```

Comment:
I might rephrase this to 
```suggestion
        it('should not allow to add a token without an oracle price', async () => {
```